tests/advanced_visibility_test.go

```diff
@@ -103,15 +102,9 @@ func (s *AdvancedVisibilitySuite) SetupSuite() {
 		// Allow the scavenger to remove any build ID regardless of when it was last default for a set.
 		dynamicconfig.RemovableBuildIdDurationSinceDefault.Key(): time.Microsecond,
 	}
+	s.FunctionalTestSuite.SetupSuiteWithDefaultCluster(testcore.WithDynamicConfigOverrides(dynamicConfigOverrides))
 
-	if testcore.UsingSQLAdvancedVisibility() {
-		s.FunctionalTestSuite.SetupSuiteWithCluster("testdata/cluster.yaml", testcore.WithDynamicConfigOverrides(dynamicConfigOverrides))
-		s.Logger.Info(fmt.Sprintf("Running advanced visibility test with %s/%s persistence", testcore.TestFlags.PersistenceType, testcore.TestFlags.PersistenceDriver))
-		s.isElasticsearchEnabled = false
-	} else {
-		s.FunctionalTestSuite.SetupSuiteWithCluster("testdata/es_cluster.yaml", testcore.WithDynamicConfigOverrides(dynamicConfigOverrides))
-		s.Logger.Info("Running advanced visibility test with Elasticsearch persistence")
-		s.isElasticsearchEnabled = true
+	if !testcore.UseSQLVisibility() {
```

Comment:
It feels a little weird that suites are calling the global function `testcore.UseSQLVisibility`... I would expect/prefer something more like:

There's one function in the code that examines flags/env vars to get persistence + vis config. TestCluster or maybe whatever calls TestCluster calls that function, then saves the results in TestCluster. The base suite has methods that read the persistence + vis type from there.

(Doesn't have to be this PR)

Reply:
This is a great idea and I was thinking about something similar. For now I am removing unnecessary indirections.